multimodal/src/autogluon/multimodal/matcher.py

```diff
@@ -1144,62 +1255,68 @@ def evaluate(
         else:
             return results
 
-    def semantic_search(
+    def evaluate(
         self,
-        query_data: Union[pd.DataFrame, dict, list],
-        response_data: Union[pd.DataFrame, dict, list],
-        id_mappings: Dict[str, Dict],
-        return_prob: Optional[bool] = False,
-        as_pandas: Optional[bool] = None,
+        data: Optional[Union[pd.DataFrame, dict, list]] = None,
+        query_data: Optional[list] = None,
+        response_data: Optional[list] = None,
+        id_mappings: Optional[Dict[str, Dict]] = None,
+        metrics: Optional[Union[str, List[str]]] = None,
+        return_pred: Optional[bool] = False,
+        chunk_size: Optional[int] = 1024,
+        cosine: Optional[bool] = True,
+        top_k: Optional[int] = 100,
     ):
         """
-        Perform a cosine similarity search between query data and response data.
+        Evaluate model on a test dataset.
 
         Parameters
         ----------
+        data
+            A dataframe, containing the same columns as the training data.
         query_data
-            The query data.
+            A list of queries.
         response_data
-            The response data.
+            A list of response data, in which to match the queries.
         id_mappings
```

Comment:
We can later try to rename id_mappings as `linked_content`, and support DataFrame as a unified data storage format.

Reply:
Sure. marked a TODO in the `fit()` arguments.